### ISSUE TITLE
Is there Infix search possible

### ISSUE THREAD

**Issue by nhaberl** [wolfgarbe/PruningRadixTrie#4]
Is there a chance also to look for **croso** to get results for microsoft?
So not only Prefix search.

**Comment by wolfgarbe** [comment]
With the current PruningRadixTrie structure there is no infix search possible, unless you want to exhaustively scan the whole trie. It would be possible to  modify the structure and the algorithm though: When adding a term to the trie you could also add all the suffixes of the term to the trie. But this would significantly increase memory consumption and lookup time.

**Comment by nhaberl** [comment]
Ok, arent you already indexing all single characters?
So index size should be the same ?
Why is it heavier to search for chractrrs inside the indexed terms ? 

Sorry if I am totally wrong but werent Tries invented to search anywhere ?

You can look here, maybe its somehow important especially the Suffix Trie
https://github.com/npgall/concurrent-trees/blob/master/README.md

**Comment by wolfgarbe** [comment]
> Sorry if I am totally wrong but weren't Tries invented to search anywhere ?

No, Radix tries (prefix tries) are supporting efficient prefix search, and Suffix tries are supporting efficient suffix search, None of them is supporting efficient substring (neither prefix nor suffix) search.

**Comment by nhaberl** [comment]
Ok thanks, so something similar to the link I've added is not possible anywhere soon ?

**Comment by wolfgarbe** [comment]
I have currently no plans to implement it. But anybody is free to fork the repository and add a substring lookup functionality.
